pkg/yttlibrary/overlay/array.go

```diff
@@ -34,9 +34,12 @@ func (o Op) mergeArrayItem(
 	}
 
 	for _, leftIdx := range leftIdxs {
-		replace, err := o.apply(leftArray.Items[leftIdx].Value, newItem.Value, matchChildDefaults)
-		if err != nil {
-			return err
+		replace := true
+		if leftArray.Items[leftIdx].Value != nil {
+			replace, err = o.apply(leftArray.Items[leftIdx].Value, newItem.Value, matchChildDefaults)
```

Comment:
it seems that apply is able to indicate when replace is necessary. any reason why not check nil value for left side within apply?

Reply:
I had been thinking keeping this behavior specifically about the merge operation.

I'll take a second look.

Reply:
I see. What was subtle to me was that:
- only the merge and assert operations continue the traversal
- only the merge operation is sensitive to `replace`

I like the simplicity of the null check in `apply()`.

👍🏼

Reply:
this recursion does get a little mind bending. it's a little weird since inside apply we dont really look at left as a primary thing.

im starting to like your initial approach more since it doesnt raise a question inside apply why are we exiting early and acting on left instead of right.

Reply:
> this recursion does get a little mind bending

Well, I'm glad to hear I'm not alone in that experience! 😌 

> im starting to like your initial approach more since it doesnt raise a question inside apply why are we exiting early and acting on left instead of right.

I find the first approach more readable and the second approach tighter.  In this case (and given that the probability of adding more operations to overlays is low, so reading is starting to become the overwhelmingly primary activity, here), I vote for the former.

Reply:
👍 let's roll with original approach.